cmd/agent/app/agent.go

```diff
@@ -84,6 +84,24 @@ func run(ctx context.Context, karmadaConfig karmadactl.KarmadaConfig, opts *opti
 	}
 	controlPlaneRestConfig.QPS, controlPlaneRestConfig.Burst = opts.KubeAPIQPS, opts.KubeAPIBurst
 
+	clusterConfig, err := controllerruntime.GetConfig()
+	if err != nil {
+		return fmt.Errorf("error building kubeconfig of member cluster: %s", err.Error())
+	}
+	clusterKubeClient := kubeclientset.NewForConfigOrDie(clusterConfig)
+	// ensure namespace where the impersonation secret to be stored in member cluster.
+	if _, err = util.EnsureNamespaceExist(clusterKubeClient, opts.ClusterNamespace, false); err != nil {
+		return err
+	}
+
+	// create a ServiceAccount for impersonation in cluster.
+	impersonationSA := &corev1.ServiceAccount{}
+	impersonationSA.Namespace = opts.ClusterNamespace
+	impersonationSA.Name = names.GenerateServiceAccountName("impersonator")
```

Comment:
> This will block people deploy more than one `karmada-agent` on the same cluster since the `impersonator` is in conflict.

Sorry, the SA is created on the member cluster, so there will be only one SA on a cluster.